module/VuFind/src/VuFind/Auth/Manager.php

```diff
@@ -754,6 +773,25 @@ public function login($request)
                 throw new AuthException('authentication_error_technical', 0, $e);
             }
 
+            // Attempt catalog login so that any bad credentials are cleared before further processing
+            // (avoids e.g. multiple login attempts by account AJAX checks):
+            if ($this->ilsAuthenticator && $username = $user->getCatUsername()) {
```

Comment:
Is it possible we want to make this check configurable? At very least, should we make sure we have an ILS and that it's not offline before we start clearing credentials?

Reply:
I added the offline check. Otherwise this is exactly the same as what happens when you go to any page that calls `storedCatalogLogin`, just done earlier.

Reply:
Wait, no, it's not exactly the same since storedCatalogLogin() doesn't persist the cleared credentials. You're right, this a bit more...

Reply:
Okay, added more preconditions and a setting that can be used disable this.